openedx/core/djangoapps/schedules/management/commands/tests/test_send_course_update.py

```diff
@@ -0,0 +1,40 @@
+from mock import patch
+from unittest import skipUnless
+
+from django.conf import settings
+
+from openedx.core.djangoapps.schedules import resolvers, tasks
+from openedx.core.djangoapps.schedules.management.commands import send_course_update as nudge
+from openedx.core.djangoapps.schedules.management.commands.tests.send_email_base import ScheduleSendEmailTestBase
+from openedx.core.djangoapps.schedules.management.commands.tests.upsell_base import ScheduleUpsellTestMixin
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+
+
+@skip_unless_lms
+@skipUnless(
+    'openedx.core.djangoapps.schedules.apps.SchedulesConfig' in settings.INSTALLED_APPS,
+    "Can't test schedules if the app isn't installed",
+)
+class TestSendCourseUpdate(ScheduleUpsellTestMixin, ScheduleSendEmailTestBase):
+    __test__ = True
+
+    # pylint: disable=protected-access
+    tested_resolver = resolvers.CourseUpdateResolver
+    tested_task = tasks.ScheduleCourseUpdate
+    deliver_task = tasks._course_update_schedule_send
+    tested_command = nudge.Command
+    deliver_config = 'deliver_course_update'
+    enqueue_config = 'enqueue_course_update'
+    expected_offsets = xrange(-7, -77, -7)
+
+    queries_deadline_for_each_course = True
+
+    def setUp(self):
+        super(TestSendCourseUpdate, self).setUp()
+        self.patcher = patch('openedx.core.djangoapps.schedules.resolvers.get_week_highlights')
+        self.mock_highlights = self.patcher.start()
+        self.mock_highlights.return_value = ['Highlight {}'.format(num + 1) for num in range(3)]
+
+    def tearDown(self):
+        self.patcher.stop()
```

Comment:
not: could you do `self.addCleanup(self.patcher.stop)` instead of defining a tearDown?

Reply:
yes.  do you consider that a better practice?

Reply:
I agree that it's better to keep the patching consolidated.  And, no longer need to store it as a member variable.  Updated.  Thanks for the suggestion.